cmd/soroban-cli/src/events.rs

```diff
@@ -0,0 +1,437 @@
+
+        let events = match self.rpc_url.as_ref() {
+            Some(rpc_url) => self.run_against_rpc_server(rpc_url).await?,
+            _ => match self.events_file.as_ref() {
+                Some(path) => self.run_in_sandbox(path)?,
+                _ => return Err(Error::TargetRequired),
+            },
+        };
+
+        for event in &events {
+            match self.output {
+                // Should we pretty-print the JSON like we're doing here or just
+                // dump an event in raw JSON on each line? The latter is easier
+                // to consume programmatically.
+                OutputFormat::Json => {
+                    println!(
+                        "{}",
+                        serde_json::to_string_pretty(&event).map_err(|e| {
+                            Error::InvalidJson {
+                                debug: format!("{:#?}", event),
+                                error: e,
+                            }
+                        })?,
+                    );
+                }
+                OutputFormat::Plain => print_event(event)?,
+                OutputFormat::Pretty => pretty_print_event(event)?,
+            }
+        }
+
+        Ok(())
+    }
+
+    async fn run_against_rpc_server(&self, rpc_url: &str) -> Result<Vec<Event>, Error> {
+        if self.start_ledger == 0 && self.end_ledger == 0 {
+            return Err(Error::LedgerRangeRequired);
+        }
+
+        let client = Client::new(rpc_url);
+        Ok(client
+            .get_events(
+                self.start_ledger,
+                self.end_ledger,
+                Some(self.event_type),
+                &self.contract_ids,
+                &self.topic_filters,
+                Some(self.count),
+            )
+            .await?
+            .unwrap_or_default())
+    }
+
+    fn run_in_sandbox(&self, path: &path::PathBuf) -> Result<Vec<Event>, Error> {
+        if !path.exists() {
+            return Err(Error::InvalidFile {
+                path: path.to_str().unwrap().to_string(),
+            });
+        }
+
+        let count: usize = if self.count == 0 {
+            std::usize::MAX
+        } else {
+            self.count
+        };
+
+        // Read the JSON events from disk and find the ones that match the
+        // contract ID filter(s) that were passed in.
+        Ok(snapshot::read_events(path)
+            .map_err(|err| Error::CannotReadFile {
+                path: path.to_str().unwrap().to_string(),
+                error: err.to_string(),
+            })?
+            .iter()
+            // FIXME: We assume here that events are read off-disk in
+            // chronological order, but we should probably be sorting by ledger
+            // number (and ID, for events within the same ledger), instead,
+            // though it's likely that this logic belongs more in
+            // `snapshot::read_events()`.
+            .rev()
+            // The ledger range is optional in sandbox mode.
```

Comment:
*style*
I'd move this comment down a line so it is next to the relevant check, as I missed it initially.